Error installing Rust toolchain with RubyGems

Hi, I wanted to report an error installing the Rust toolchain on Mac ARM (which happens by default if users don't have Rust installed). To reproduce:

```sh
RB_SYS_FORCE_INSTALL_RUST_TOOLCHAIN=true gem install wasmtime --platform ruby
```

Output

```text
Building native extensions. This could take a while...
ERROR:  Error installing wasmtime:
    ERROR: Failed to build gem native extension.

    current directory: /Users/user/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/wasmtime-6.0.1/ext
/Users/user/.rbenv/versions/3.2.1/bin/ruby -I /Users/user/.rbenv/versions/3.2.1/lib/ruby/3.2.0 extconf.rb
checking for cargo... no

current directory: /Users/user/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/wasmtime-6.0.1/ext
make DESTDIR\= sitearchdir\=./.gem.20230328-15890-3en8ju sitelibdir\=./.gem.20230328-15890-3en8ju clean

current directory: /Users/user/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/wasmtime-6.0.1/ext
make DESTDIR\= sitearchdir\=./.gem.20230328-15890-3en8ju sitelibdir\=./.gem.20230328-15890-3en8ju
info: downloading installer
info: profile set to 'minimal'
info: default host triple is aarch64-apple-darwin
info: skipping toolchain installation


Rust is installed now. Great!

To get started you need Cargo's bin directory 
(/Users/user/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/wasmtime-6.0.1/ext/.rb-sys/stable/cargo/bin)
in your PATH
environment variable. This has not been done automatically.

To configure your current shell, run:
source 
"/Users/user/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/wasmtime-6.0.1/ext/.rb-sys/stable/cargo/env"
make: rustup: No such file or directory
make: *** [/Users/user/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/wasmtime-6.0.1/ext/.rb-sys/stable/cargo/bin/cargo] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/user/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/wasmtime-6.0.1 for inspection.
Results logged to /Users/user/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/extensions/arm64-darwin-22/3.2.0/wasmtime-6.0.1/gem_make.out
```

Ref: https://github.com/ankane/tokenizers-ruby/issues/28

--- a/gem/lib/rb_sys/mkmf.rb
+++ b/gem/lib/rb_sys/mkmf.rb
@@ -244,8 +244,8 @@ module RbSys
         $(CARGO):
         \t$(Q) $(MAKEDIRS) $(CARGO_HOME) $(RUSTUP_HOME)
         \t$(Q) curl --proto '=https' --tlsv1.2 --retry 10 --retry-connrefused -fsSL "https://sh.rustup.rs" | sh -s -- --no-modify-path --profile $(RB_SYS_RUSTUP_PROFILE) --default-toolchain none -y
-        \t$(Q) rustup toolchain install $(RB_SYS_DEFAULT_TOOLCHAIN) --profile $(RB_SYS_RUSTUP_PROFILE)
-        \t$(Q) rustup default $(RB_SYS_DEFAULT_TOOLCHAIN)
+        \t$(Q) $(CARGO_HOME)/bin/rustup toolchain install $(RB_SYS_DEFAULT_TOOLCHAIN) --profile $(RB_SYS_RUSTUP_PROFILE)
+        \t$(Q) $(CARGO_HOME)/bin/rustup default $(RB_SYS_DEFAULT_TOOLCHAIN)
 
         $(RUSTLIB): $(CARGO)
         #{endif_stmt}

